In [34]:
import pandas as pd 
import numpy as np 


## EDA

In [3]:
df = pd.read_csv("athlete_events.csv")
df

ID                      Name Sex   Age  Height  Weight  \
0            1                 A Dijiang   M  24.0   180.0    80.0   
1            2                  A Lamusi   M  23.0   170.0    60.0   
2            3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
3            4      Edgar Lindenau Aabye   M  34.0     NaN     NaN   
4            5  Christine Jacoba Aaftink   F  21.0   185.0    82.0   
...        ...                       ...  ..   ...     ...     ...   
271111  135569                Andrzej ya   M  29.0   179.0    89.0   
271112  135570                  Piotr ya   M  27.0   176.0    59.0   
271113  135570                  Piotr ya   M  27.0   176.0    59.0   
271114  135571        Tomasz Ireneusz ya   M  30.0   185.0    96.0   
271115  135571        Tomasz Ireneusz ya   M  34.0   185.0    96.0   

                  Team  NOC        Games  Year  Season            City  \
0                China  CHN  1992 Summer  1992  Summer       Barcelona   
1                China  CHN  2012 Summer  2012  Summer          London   
2              Denmark  DEN  1920 Summer  1920  Summer       Antwerpen   
3       Denmark/Sweden  DEN  1900 Summer  1900  Summer           Paris   
4          Netherlands  NED  1988 Winter  1988  Winter         Calgary   
...                ...  ...          ...   ...     ...             ...   
271111        Poland-1  POL  1976 Winter  1976  Winter       Innsbruck   
271112          Poland  POL  2014 Winter  2014  Winter           Sochi   
271113          Poland  POL  2014 Winter  2014  Winter           Sochi   
271114          Poland  POL  1998 Winter  1998  Winter          Nagano   
271115          Poland  POL  2002 Winter  2002  Winter  Salt Lake City   

                Sport                                     Event Medal  
0          Basketball               Basketball Men's Basketball   NaN  
1                Judo              Judo Men's Extra-Lightweight   NaN  
2            Football                   Football Men's Football   NaN  
3          Tug-Of-War               Tug-Of-War Men's Tug-Of-War  Gold  
4       Speed Skating          Speed Skating Women's 500 metres   NaN  
...               ...                                       ...   ...  
271111           Luge                Luge Mixed (Men)'s Doubles   NaN  
271112    Ski Jumping  Ski Jumping Men's Large Hill, Individual   NaN  
271113    Ski Jumping        Ski Jumping Men's Large Hill, Team   NaN  
271114      Bobsleigh                      Bobsleigh Men's Four   NaN  
271115      Bobsleigh                      Bobsleigh Men's Four   NaN  

[271116 rows x 15 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 271116 entries, 0 to 271115
Data columns (total 15 columns):
 #   Column  Non-Null Count   Dtype  
---  ------  --------------   -----  
 0   ID      271116 non-null  int64  
 1   Name    271116 non-null  object 
 2   Sex     271116 non-null  object 
 3   Age     261642 non-null  float64
 4   Height  210945 non-null  float64
 5   Weight  208241 non-null  float64
 6   Team    271116 non-null  object 
 7   NOC     271116 non-null  object 
 8   Games   271116 non-null  object 
 9   Year    271116 non-null  int64  
 10  Season  271116 non-null  object 
 11  City    271116 non-null  object 
 12  Sport   271116 non-null  object 
 13  Event   271116 non-null  object 
 14  Medal   39783 non-null   object 
dtypes: float64(3), int64(2), object(10)
memory usage: 31.0+ MB


In [6]:
df.describe()

ID            Age         Height         Weight  \
count  271116.000000  261642.000000  210945.000000  208241.000000   
mean    68248.954396      25.556898     175.338970      70.702393   
std     39022.286345       6.393561      10.518462      14.348020   
min         1.000000      10.000000     127.000000      25.000000   
25%     34643.000000      21.000000     168.000000      60.000000   
50%     68205.000000      24.000000     175.000000      70.000000   
75%    102097.250000      28.000000     183.000000      79.000000   
max    135571.000000      97.000000     226.000000     214.000000   

                Year  
count  271116.000000  
mean     1978.378480  
std        29.877632  
min      1896.000000  
25%      1960.000000  
50%      1988.000000  
75%      2002.000000  
max      2016.000000

In [42]:
print(f"The number of rows are: {df.shape[0]}")
print(f"The number of columns are: {df.shape[1]}")

The number of rows are: 271116
The number of columns are: 15


### Data Types

In [11]:
df.dtypes

ID          int64
Name       object
Sex        object
Age       float64
Height    float64
Weight    float64
Team       object
NOC        object
Games      object
Year        int64
Season     object
City       object
Sport      object
Event      object
Medal      object
dtype: object

<div class="alert alert-block alert-info">

Befor we delve into the data here is a quick overview of what we are going to be dealing with. 

This is a historical dataset on the modern Olympic Games, including all the Games from Athens 1896 to Rio 2016. It includes 15 features as show below. 

`ID (int64):` Unique number for each athlete.

`Name (object):`  Athlete's name.

`Sex (object):` Sex of athete M: male or W:women

`Age (float64):` Age of athlete.

`Height (float64 - cm):` Height of athlete in cm.

`Weight (float64 - kg):` Weight of athlete in cm.

`Team (object):` Team country of athelete. 

`NOC (object):` National Olympic Committee by each country.

`Games (object):` Year and season of the Olyimic Game.

`Year (int64):` Year the athlete competed.

`Season (object):` Season the Olymics were played. 

`City (object):` The city where the Olymics were hosted. 

`Sport (object):` The sport the athelete competed in. 

`Event (object):` The name of the event the athelete competed in. 

`Medal (object):` The medal the athelete won in the competition Gold, Silver, Bronze, or NA.






</div>

<div class="alert alert-block alert-info">

| Column Name       | Description                                                               | Data Type     |
| ----------------- | ------------------------------------------------------------------------- | ------------- |
| ID                | Unique number for each athlete                                            | int           |
| Name              | Athlete's name.                                                           | object        |
| Sex               | Athete's sex  M: male or W: women                                         | object        |
| Age               | Athete's age                                                              | float         |
| Height            | Athete's height in cm                                                     | float         |
| Weight            | Athete's weight in kg                                                     | float         |
| Team              | Athelete's team country                                                   | object        |
| NOC               | National Olympic Committee by each country.                               | object        |
| Games             | Year and season of the Olyimic Game.                                      | object        |
| Year              | Year the athlete competed.                                                | int           | 
| Season            | Season the Olymics were played.                                           | object        |
| City              | The city where the Olymics were hosted.                                   | object        |
| Sport             | The sport the athelete competed in                                        | object        |
| Event             | The name of the event the athelete competed in                            | object        | 
| Medal             | The medal the athelete won in the competition Gold, Silver, Bronze, or NA.| object        |                         







</div>

Lets check the unique values in each feature. 

In [33]:
column = df.columns.to_list()

for column_name in column:
    df[column_name].unique()
    print(f"The column: {column_name} has {len(df[column_name].unique())} unique values that are \n {df[column_name].unique()}\n")



The column: ID has 135571 unique values that are 
 [     1      2      3 ... 135569 135570 135571]

The column: Name has 134732 unique values that are 
 ['A Dijiang' 'A Lamusi' 'Gunnar Nielsen Aaby' ... 'Andrzej ya' 'Piotr ya'
 'Tomasz Ireneusz ya']

The column: Sex has 2 unique values that are 
 ['M' 'F']

The column: Age has 75 unique values that are 
 [24. 23. 34. 21. 25. 27. 31. 33. 18. 26. 22. 30. 32. 28. 54. 20. 17. 43.
 47. 29. 41. 45. 49. 53. 57. nan 19. 38. 35. 16. 37. 15. 42. 46. 40. 36.
 14. 39. 48. 52. 44. 55. 50. 71. 63. 51. 58. 13. 60. 75. 65. 56. 64. 68.
 84. 12. 72. 59. 61. 70. 74. 62. 67. 69. 73. 66. 11. 76. 88. 96. 80. 10.
 81. 77. 97.]

The column: Height has 96 unique values that are 
 [180. 170.  nan 185. 188. 183. 168. 186. 182. 172. 159. 171. 184. 175.
 189. 176. 163. 187. 169. 174. 190. 194. 177. 167. 164. 198. 165. 156.
 179. 178. 166. 181. 155. 160. 173. 192. 195. 191. 203. 158. 154. 153.
 152. 193. 162. 197. 157. 207. 147. 150. 202. 161. 210. 205. 200. 148.
 

Some of the features' data type can be changed to categorical, but for now I will leave it as it and may come back to this. 

### Missing Values

In [37]:
df.isnull().any()

ID        False
Name      False
Sex       False
Age        True
Height     True
Weight     True
Team      False
NOC       False
Games     False
Year      False
Season    False
City      False
Sport     False
Event     False
Medal      True
dtype: bool

There are 4 features that have missing values: `Age`, `Height`, `Weight`, and `Medal`

In [46]:
print(f"The number of missing values for the columns are as follows: \n{df.isnull().sum()}")

The number of missing values for the columns are as follows: 
ID             0
Name           0
Sex            0
Age         9474
Height     60171
Weight     62875
Team           0
NOC            0
Games          0
Year           0
Season         0
City           0
Sport          0
Event          0
Medal     231333
dtype: int64


In [51]:
print(f"The percentage distribution for each column is as follows: {round((df.isnull().sum()/df.shape[0])*100,2)}")

The percentage distribution for each column is as follows: ID         0.00
Name       0.00
Sex        0.00
Age        3.49
Height    22.19
Weight    23.19
Team       0.00
NOC        0.00
Games      0.00
Year       0.00
Season     0.00
City       0.00
Sport      0.00
Event      0.00
Medal     85.33
dtype: float64
